### DISCUSSION TITLE
Image Resolution Options

### DISCUSSION THREAD

**Discussion by superskirv** [LostRuins/koboldcpp#777] [Ideas]
Can we get an option in the "Styles" window for Image Resolutions? The current defaults are great for SD1.5, but SDXL and Cascade Models perform better at 1024-1280. Even if its a SDXL/Hi-Res check mark and it just doubles the current values would be nice. 512x341 would become 1024x682, which in my opinion is still too low, but at least its better. Looking into the future, its probably better to just allow users to type whatever number they want. With as quickly as things are growing we could be generating images as high as 2048x2048 as the default in the next year or two.

The current option "Save High-Res" just seems to change the file compression, which is extremely less useful.

Thank you for this easy to use build.

**Comment by LostRuins** [comment]
I'll take it into consideration, but KoboldCpp doesn't handle larger images well unfortunately. The high-res option does indeed increase the resolution of the received image, you can also specify the desired aspect ratio to use (non square will be larger). Make sure you don't run it with `clamped` or `quick` as that reduces the output res.

**Reply by the-ResQ** [reply]
First of all: the in-chat imagegen is amazing, thank you so much for that!

**Perhaps it can show a 512px "thumbnail" in the chat and on opening the image, it shows the original in best resolution and quality?**

It's a bummer the images are only 512x512. I'm running a RTX 4080 and 7800X3D, my PC has absolutely no problem generating multiple images and running the chatbot at the same time.

As it is, the imagegen is interesting, but close to unusable. The quality of the images is just too low, so currently it's just a gimmick. I hope you can give us the option to increase the resolution of the images! Even if the performance of the chatbot or UI is lower as a result, I'll accept that if the images are better.

**Reply by LostRuins** [reply]
Actually if you click on the image, it does open up a larger display. If you want high resolutions, you can try the standalone stableui that is bundled in KoboldCpp at http://localhost:5001/sdui

**Reply by the-ResQ** [reply]
Thanks for the response! :) the issue with that: it removes you out of the chat experience. It's not the same as having an image generated in the chat UI itself. It's extra steps and you'd even need to copypaste the context, removing the whole point of the image-gen-in-chat feature.
Changing your tab to the SDUI and creating an image there is nothing I need koboldcpp for, I could already do that with automatic1111 before :) thanks for your work!!

**Comment by Amin456789** [comment]
indeed, an resoluton option in style will be great,to choose something like 384x256 for sdxl turbo for faster generation

**Comment by digital-garbage** [comment]
Hi! This one is still an actual question for me since I have an RTX 3090 and can run both LLM and a good SDXL model at the same time - I just want to increase the resolution, but can't find any option for it, will it be added in a future release? Thanks.

**Comment by lorenzolamasse** [comment]
Agree on this, it would improve the experience furthermore. As well, but I know it's more work, being able to use an adetailer for face improvement would be nice as that's the main part that is clearly lowres...
